### PR TITLE
Remove run test subcommand and --cleanup flag, print test status

### PR DESCRIPTION
**Pull request by jlegrone**
**What this PR does / why we need it**:

Related to #6020. This PR removes the `helm test run` subcommand and `--cleanup` flag. The output of `helm test --help` is now:

```
The test command runs the tests for a release.

The argument this command takes is the name of a deployed release.
The tests to be run are defined in the chart that was installed.

Usage:
  helm test [RELEASE] [flags]

Flags:
  -h, --help               help for test
      --timeout duration   time to wait for any individual Kubernetes operation (like Jobs for hooks) (default 5m0s)
```

More output has also been added during execution of `helm test`. The output of `helm test <release>` now looks like:

```
Pod nginx-test-bar pending
Pod nginx-test-bar pending
Pod nginx-test-bar succeeded
Pod nginx-test-foo pending
Pod nginx-test-foo pending
Pod nginx-test-foo succeeded
TEST SUITE:     nginx-test-foo
Last Started:   2019-09-05 15:04:12.33368 -0400 EDT m=+5.107841576
Last Completed: 2019-09-05 15:04:14.147617 -0400 EDT m=+6.921723894
Phase:          Succeeded
TEST SUITE:     nginx-test-bar
Last Started:   2019-09-05 15:04:08.31585 -0400 EDT m=+1.090131819
Last Completed: 2019-09-05 15:04:12.314482 -0400 EDT m=+5.088644403
Phase:          Succeeded
```

For now in order to perform test resource cleanup, users will need to set `helm.sh/hook-delete-policy`, manually delete resources with kubectl, or set [TTL for Job resources](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/). The hook labels proposal in https://github.com/helm/helm/issues/6365 should improve on the ergonomics of using kubectl for cleanup if we don't implement something directly in helm first.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/release_testing.go [hunk 1]
```diff
@@ -18,28 +18,39 @@ package main
...
 
+	"helm.sh/helm/cmd/helm/require"
 	"helm.sh/helm/pkg/action"
 )
 
 const releaseTestHelp = `
-The test command consists of multiple subcommands around running tests on a release.
-
-Example usage:
-    $ helm test run [RELEASE]
```
Comment (marckhouzam) [inline cmd/helm/release_testing.go]: Currently `helm test run <TAB>` will provide dynamic completion by listing available releases.

If you remove the `run` subcommand, could you update the dynamic completion code by changing the string `helm_test_run` to `helm_test` in `cmd/helm/root.go`?

Thank you.

Comment (jlegrone) [reply cmd/helm/release_testing.go]: Nice catch @marckhouzam, updated.

**Review by technosophos — APPROVED** [review]
Code LGTM

**Review by scottrigby — APPROVED** [review]
haven't tested, but look good. also yay for simplifying 💯

**Comment by bacongobbler** [comment]
If at all possible, can you (the next core maintainer to review this PR) please hold on merging this until I get back next week? I want to take a closer look at the ramifications of removing the `--cleanup` flag when it comes to the user experience. I want to make sure we have those user stories captured and well-understood before we decide to remove a fairly significant feature of `helm test` in favour of an alternative approach (hook annotations).

**Comment by unguiculus** [comment]
If I understand it correctly, #3211 is fixed in v3?

**Comment by jlegrone** [comment]
> If I understand it correctly, #3211 is fixed in v3?

Yes, I wasn't aware of that issue but it should be closed by https://github.com/helm/helm/pull/6054.

File: cmd/helm/release_testing.go [hunk 2]
```diff
@@ -18,28 +18,39 @@ package main
 
 import (
 	"io"
+	"time"
 
 	"github.com/spf13/cobra"
 
+	"helm.sh/helm/cmd/helm/require"
```
Comment (mattfarina) [inline cmd/helm/release_testing.go]: Helm moved to Go modules. That means the major version needs to be in the import path. This applies when something has a major version of 2 or higher. So, the helm import paths need to have a /v3 in them. For example,

`"helm.sh/helm/cmd/helm/require"` -> `"helm.sh/helm/v3/cmd/helm/require"`

File: pkg/action/release_testing.go [hunk 3]
```diff
@@ -45,39 +40,21 @@ func NewReleaseTesting(cfg *Configuration) *ReleaseTesting {
...
-		return err
+		return rel, err
 	}
 
 	if err := r.cfg.execHook(rel, release.HookTest, r.Timeout); err != nil {
 		r.cfg.Releases.Update(rel)
-		return err
-	}
-
-	if r.Cleanup {
```
Comment (bacongobbler) [inline pkg/action/release_testing.go]: First off, thank you for taking the time to refactor `helm test`. We really appreciate the work you've put into this.

That being said, we cannot remove this feature and replace it with hook annotations. This feature serves a very different purpose than what the hook annotations provide, which is why `helm test` was not originally designed around the hook system.

A very simple test where this flag is required:

1. I have a chart that someone else has built (e.g. `stable/wordpress`)
1. I want to run the tests
1. I want to have the test hooks cleaned up in preparation for the next test run

Prior to this change, the use case was done using the following commands:

```
$ helm install wordpress stable/wordpress
$ helm test run wordpress --cleanup
```

Now with this functionality removed, a user must now perform the following:

```
$ helm fetch stable/wordpress --untar
$ $EDITOR wordpress/templates/tests/
$ helm install wordpress ./wordpress
$ helm test wordpress
```

So now, the user must MODIFY the chart before installing to handle the use case. If they don't want to clean up test hooks after the test run, then they have to modify the chart AGAIN, run `helm upgrade` and finally run `helm test`. This is a massive barrier of entry for users who just want their test hooks to be cleaned up after running `helm test`. It's for that reason this functionality must remain as-is. I don't see a way around this with the way hook annotations work today, sadly.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
All that being said, once I rebased this PR on master, everything seems to work as described. Thank you!

File: pkg/action/release_testing.go (hunk 3 above)
Comment (jlegrone) [reply pkg/action/release_testing.go]: Perhaps we should schedule some time to discuss this; I think the workflow concerns here are addressed by https://github.com/helm/helm/issues/6365.

When those changes are combined with https://github.com/helm/helm/issues/6364 and the proposal to amend the default `hook-delete-policy` from https://github.com/helm/helm/issues/6020 I think we should be able to deliver a much nicer experience than Helm 2 out of the box. This PR is not representative of the whole picture.

**Review by bacongobbler — COMMENTED** [review]
After discussion on the dev call, we have a path forward and this is a step in that direction. Once this has been rebased and @mattfarina's comments have been addressed we can get this one in. Thanks!

File: pkg/action/release_testing.go (hunk 3 above)
Comment (bacongobbler) [reply pkg/action/release_testing.go]: Discussed offline. Workflow concerns are indeed addressed through other proposed paths.

**Review by bacongobbler — APPROVED** [review]
LGTM! Will need another maintainer to sign off since this is a `size/L`

**Comment by bacongobbler** [comment]
Oh! I see @technosophos LGTM'd this change a while back. Merging!

**Pull request merged**